lib/Parser/Parser.cpp

```diff
@@ -244,6 +246,10 @@ Token Lexer::getNextToken(std::string &ErrStr) {
         return Token{Token::Error, SignBegin, 0, APInt()};
       }
     }
+    while (!KnownBitsFlag && !MoreKnownBitsFlag && !NumSignBitsFlag && (*Begin == '?' || *Begin == 'd')) {
```

Comment:
this line is weird. when you put conditions in a loop test, you are implying that they may change as the loop iterates --  otherwise, why put them in the loop test? but the first three parts of this loop test cannot change as the loop executes